src/viking.js

```diff
@@ -1,11 +1,99 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health
+        this.strength = strength
+    }
+    attack() {
+        return this.strength
+    }
+    receiveDamage(damage) {
+        this.health -= damage
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor (name, health, strength) {
+        super(health, strength)
+        this.name = name
+    }
+    receiveDamage(damage) {
+        super.receiveDamage(damage)
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+        } else {
+            return `${this.name} has died in act of combat`
+        }
+    }
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+}
+
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        super.receiveDamage(damage)
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+        } else {
+            return `A Saxon has died in combat`
+        }
+    }
+}
+
+
 
 // War
-class War {}
+class War {
+    vikingArmy = [];
+  saxonArmy = [];
+  addViking(Viking) {
+    this.vikingArmy.push(Viking);
+  }
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+  vikingAttack() {
+    let randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    let randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+
+    let result = randomSaxon.receiveDamage(randomViking.strength);
+
+    for (let i = 0; i < this.saxonArmy.length; i++) {
```

Comment:
Here, you are running a loop to check if there's "dead" saxons and removing them from the array. The thing is, you are applying a `vikingAttack()` method, to target a random saxon, but since you have it's random index, it will be that same index you'll use to check if the `health` is smaller or equal to `0` and for you to remove it from the `saxonArmy` array. 
So, there's no need for this 'extra' step of going trough all the array and finding the ones who are dead.